Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -578,72 +560,55 @@ public enum DefaultFunctionOpacityOptions {
         { EnforceDeterminism, OptionCompatibility.CheckOptionLocalImpliesLibrary },
-        { RelaxDefiniteAssignment, OptionCompatibility.CheckOptionLibraryImpliesLocal },
-        { AllowAxioms, OptionCompatibility.CheckOptionLibraryImpliesLocal },
-        { AllowWarnings, (reporter, origin, prefix, option, localValue, libraryValue) => {
-            if (OptionCompatibility.OptionValuesImplied(libraryValue, localValue)) {
-              return true;
-            }
-            string message = OptionCompatibility.LocalImpliesLibraryMessage(prefix, option, localValue, libraryValue);
-            reporter.Warning(MessageSource.Project, ResolutionErrors.ErrorId.none, origin, message);
-            return false;
-          }
-        }
+        { RelaxDefiniteAssignment, OptionCompatibility.OptionLibraryImpliesLocalError },
+        { AllowAxioms, OptionCompatibility.OptionLibraryImpliesLocalError },
+        { AllowWarnings, OptionCompatibility.OptionLibraryImpliesLocalWarning },
+        { AllowDeprecation, OptionCompatibility.OptionLibraryImpliesLocalWarning },
+        { WarnShadowing, OptionCompatibility.OptionLibraryImpliesLocalWarning },
+        { UseStandardLibraries, OptionCompatibility.OptionLibraryImpliesLocalError },
       }
     );
-    DooFile.RegisterNoChecksNeeded(
-      WarnAsErrors,
-      ProgressOption,
-      LogLocation,
-      LogLevelOption,
-      ManualTriggerOption,
-      ShowHints,
-      Check,
-      Libraries,
-      Output,
-      PluginOption,
-      Prelude,
-      Target,
-      Verbose,
-      AllowDeprecation,
-      FormatPrint,
-      JsonDiagnostics,
-      QuantifierSyntax,
-      SpillTranslation,
-      StdIn,
-      TestAssumptions,
-      WarnShadowing,
-      ManualLemmaInduction,
-      TypeInferenceDebug,
-      GeneralTraits,
-      GeneralNewtypes,
-      TypeSystemRefresh,
-      VerificationLogFormat,
-      VerifyIncludedFiles,
-      DisableNonLinearArithmetic,
-      NewTypeInferenceDebug,
-      UseBaseFileName,
-      WarnMissingConstructorParenthesis,
-      UseJavadocLikeDocstringRewriterOption,
-      IncludeRuntimeOption,
-      InternalIncludeRuntimeOptionForExecution,
-      WarnContradictoryAssumptions,
-      WarnRedundantAssumptions,
-      VerificationCoverageReport,
-      NoTimeStampForCoverageReport,
-      DefaultFunctionOpacity,
-      UseStandardLibraries,
-      OptimizeErasableDatatypeWrapper,
-      AddCompileSuffix,
-      SystemModule,
-      ExecutionCoverageReport,
-      ExtractCounterexample,
-      ShowProofObligationExpressions
-      );
+    DooFile.RegisterNoChecksNeeded(WarnAsErrors, false);
+    DooFile.RegisterNoChecksNeeded(ProgressOption, false);
+    DooFile.RegisterNoChecksNeeded(LogLocation, false);
+    DooFile.RegisterNoChecksNeeded(LogLevelOption, false);
+    DooFile.RegisterNoChecksNeeded(ManualTriggerOption, true);
+    DooFile.RegisterNoChecksNeeded(ShowHints, false);
+    DooFile.RegisterNoChecksNeeded(Libraries, true);
```

Comment:
Aha this is the one :) I don't think we should store these paths in the doo file since it completely kills portability. It's also not actually necessary/beneficial to store them in the doo file since they aren't needed to re-parse the content, and the context consuming the doo file should know how to provide the transitive library files with potentially different paths.

Reply:
Good point. Don't recall why I set this to true.